core/CronArchive.php

```diff
@@ -1685,9 +1685,17 @@ private function getCustomDateRangeToPreProcess($idSite)
         if (is_null($cache)) {
             $cache = $this->loadCustomDateRangeToPreProcess();
         }
+
         if (empty($cache[$idSite])) {
-            return array();
+            $cache[$idSite] = array();
+        }
+
+        $customRanges = array_filter(Config::getInstance()->General['archiving_custom_ranges']);
+
+        if (!empty($customRanges)) {
+            $cache[$idSite] += $customRanges;
```

Comment:
not quite sure but wouldn'T that be an array_merge?

Reply:
yes. indeed